File: fieldmark-go/internal/domain/enums/audit_action.go
// Package enums — audit_action.go.
//
// Canonical audit-action constants for the Go stack. The persisted form in
// domain.audit_entry.action is the string value verbatim — PascalCase
// present-tense past-form per docs/reference/audit-actions.md.
//
// Source of truth: docs/reference/audit-actions.md +
// docs/reference/audit-actions.json. Adding or removing a member requires the
// Change Procedure documented there. AllAuditActions below is the iteration
// target for audit_action_conformance_test.go.
package enums

type AuditAction string

const (
	AuditActionProjectCreated                 AuditAction = "ProjectCreated"
	AuditActionProjectPlacedOnHold            AuditAction = "ProjectPlacedOnHold"
	AuditActionProjectResumed                 AuditAction = "ProjectResumed"
	AuditActionProjectClosed                  AuditAction = "ProjectClosed"
	AuditActionInspectionScheduled            AuditAction = "InspectionScheduled"
	AuditActionInspectionStarted              AuditAction = "InspectionStarted"
	AuditActionInspectionCompleted            AuditAction = "InspectionCompleted"
	AuditActionInspectionCancelled            AuditAction = "InspectionCancelled"
	AuditActionViolationOpened                AuditAction = "ViolationOpened"
	AuditActionViolationAssigned              AuditAction = "ViolationAssigned"
	AuditActionViolationVoided                AuditAction = "ViolationVoided"
	AuditActionCorrectiveActionSubmitted      AuditAction = "CorrectiveActionSubmitted"
	AuditActionCorrectiveActionTakenForReview AuditAction = "CorrectiveActionTakenForReview"
	AuditActionCorrectiveActionApproved       AuditAction = "CorrectiveActionApproved"
	AuditActionCorrectiveActionRejected       AuditAction = "CorrectiveActionRejected"
)

// AllAuditActions is the exhaustive slice of canonical action values. The
// order matches the Change Procedure list in docs/reference/audit-actions.md.
var AllAuditActions = []AuditAction{
	AuditActionProjectCreated,
	AuditActionProjectPlacedOnHold,
	AuditActionProjectResumed,
	AuditActionProjectClosed,
	AuditActionInspectionScheduled,
	AuditActionInspectionStarted,
	AuditActionInspectionCompleted,
	AuditActionInspectionCancelled,
	AuditActionViolationOpened,
	AuditActionViolationAssigned,
	AuditActionViolationVoided,
	AuditActionCorrectiveActionSubmitted,
	AuditActionCorrectiveActionTakenForReview,
	AuditActionCorrectiveActionApproved,
	AuditActionCorrectiveActionRejected,
}

File: FieldMark/FieldMark.Domain/ValueObjects/AuditAction.cs
namespace FieldMark.Domain.ValueObjects;

/// <summary>
/// Canonical audit-action enum. The persisted form in
/// <c>domain.audit_entry.action</c> is the symbol's name verbatim — PascalCase
/// present-tense past-form per <c>docs/reference/audit-actions.md</c>.
///
/// Source of truth: <c>docs/reference/audit-actions.md</c> +
/// <c>docs/reference/audit-actions.json</c> conformance fixture.
/// Adding or removing a member requires the Change Procedure documented there.
/// </summary>
public enum AuditAction
{
    ProjectCreated,
    ProjectPlacedOnHold,
    ProjectResumed,
    ProjectClosed,
    InspectionScheduled,
    InspectionStarted,
    InspectionCompleted,
    InspectionCancelled,
    ViolationOpened,
    ViolationAssigned,
    ViolationVoided,
    CorrectiveActionSubmitted,
    CorrectiveActionTakenForReview,
    CorrectiveActionApproved,
    CorrectiveActionRejected,
}

public static class AuditActionExtensions
{
    public static string AsString(this AuditAction action) => action.ToString();
}
